File: tools/pas2php-code-6/pas2php_include/fileutil.php
<?PHP
// 
// Converted From Pascal By Pas2Php v0.8
// Converts Object Pascal to PHP
// http://www.wascal.net/?page=pas2php
// Pas2Php v0.8 is Copyright (C) 2014-2016 Derek John Evans
// 
include_once('system.php');
include_once('fileutil.php');
include_once('p2pcgi.php');
include_once('phplib.php');
include_once('strutils.php');
include_once('sysutils.php');
// function FilenameIsAbsolute(const AFileName: string) : Boolean;
function FilenameIsAbsolute($AFileName) {
  // LPos : Integer;
  $LPos = 0;
  // Result : Boolean;
  $Result = false;
  $LPos = PosSet(AllowDirectorySeparators(), $AFileName);
  $Result = $LPos == 1 || $LPos > 1 && $AFileName[($LPos - 1) - 1] == DriveSeparator;
  return $Result;
}
// function FileIsReadable(const AFileName: string) : Boolean;
function FileIsReadable($AFileName) {
  // Result : Boolean;
  $Result = false;
  $Result = is_readable($AFileName);
  return $Result;
}
// function FileIsWritable(const AFileName: string) : Boolean;
function FileIsWritable($AFileName) {
  // Result : Boolean;
  $Result = false;
  $Result = is_writable($AFileName);
  return $Result;
}
// function ProgramDirectory : string;
function ProgramDirectory() {
  // Result : string;
  $Result = "";
  $Result = ExtractFilePath(CgiGetEnv(CGI_SCRIPT_FILENAME));
  return $Result;
}
// function ReadFileToString(const AFileName: string) : string;
function ReadFileToString($AFileName) {
  // Result : string;
  $Result = "";
  $Result = file_get_contents($AFileName);
  return $Result;
}
?>

File: tools/pas2php-code-6/pas2php_include/__fileutil.pas
{
 ***************************************************************************
 *                                                                         *
 *   This source is free software; you can redistribute it and/or modify   *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This code is distributed in the hope that it will be useful, but      *
 *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
 *   General Public License for more details.                              *
 *                                                                         *
 *   A copy of the GNU General Public License is available on the World    *
 *   Wide Web at <http://www.gnu.org/copyleft/gpl.html>. You can also      *
 *   obtain it by writing to the Free Software Foundation,                 *
 *   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.        *
 *                                                                         *
 ***************************************************************************
}

unit __FileUtil;

interface

uses FileUtil, P2PCgi, PhpLib, StrUtils, SysUtils;

function FilenameIsAbsolute(const AFileName: string): boolean;
function FileIsReadable(const AFileName: string): boolean;
function FileIsWritable(const AFileName: string): boolean;
function ProgramDirectory: string;
function ReadFileToString(const AFileName: string): string;

implementation

function FilenameIsAbsolute(const AFileName: string): boolean;
var
  LPos: integer;
begin
  LPos := PosSet(AllowDirectorySeparators, AFileName);
  Result := (LPos = 1) or ((LPos > 1) and (AFileName[LPos - 1] = DriveSeparator));
end;

function FileIsReadable(const AFileName: string): boolean;
begin
  Result := php_is_readable(AFileName);
end;

function FileIsWritable(const AFileName: string): boolean;
begin
  Result := php_is_writable(AFileName);
end;

function ProgramDirectory: string;
begin
  Result := ExtractFilePath(CgiGetEnv(CGI_SCRIPT_FILENAME));
end;

function ReadFileToString(const AFileName: string): string;
begin
  Result := php_file_get_contents(AFileName);
end;

end.
